### PR TITLE
feat: add eslint-plugin-ssr-friendly

### LINKED ISSUE #14283: Ensure changes made to `@carbon/react` are SSR compatible with `eslint-plugin-ssr-friendly`
We have pieces of infrastructure in place to help ensure that components are SSR compatible, for example the `useIsomorphicEffect()` internal helper hook, but we could improve this further by using static analysis to catch potential issues before they get merged.

```[tasklist]
- [ ] Add [`eslint-plugin-ssr-friendly`](https://github.com/kopiro/eslint-plugin-ssr-friendly) and fix any resulting errors
- [ ] Write docs for how to test changes for SSR compatibility (Primer has a [good example](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility) of this)
```

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #14283

#### Changelog

**New**

- add `eslint-plugin-ssr-friendly`

**Changed**

- update SliderSkeleton to set document inside useIsomorphicEffect
- add eslint ignore to test file and storybook files 

#### Testing / Reviewing

Make sure Menu and SliderSkeleton still behave as expected.

### CODE REVIEW

**Comment by tay1orjones** [comment]
I think a lot of the errors could be resolved by putting the various calls to `document` into `useIsomorphicEffect()`

File: packages/react/src/components/Menu/Menu.stories.js [hunk 1]
```diff
@@ -45,9 +45,11 @@ export const Playground = (args) => {
   const selectableOnChange = action('onChange (MenuItemSelectable)');
   const radioOnChange = action('onChange (MenuItemRadioGroup)');
 
+  // eslint-disable-next-line ssr-friendly/no-dom-globals-in-react-fc
   const target = document.getElementById('storybook-root');
```
Comment (tay1orjones) [inline packages/react/src/components/Menu/Menu.stories.js]: Honestly I'd be fine excluding `*.stories.js` altogether in the plugin config. I don't think there's a reason we need it checking the story files?

**Review by tay1orjones — APPROVED** [review]
Looks good, just one suggestion

File: packages/react/src/components/Menu/Menu.stories.js (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/Menu/Menu.stories.js]: Same thing with the `*-test.js` files actually, I don't see a reason to run this on them.

Comment (alisonjoseph) [reply packages/react/src/components/Menu/Menu.stories.js]: Yea, will make that update 👍

**Review by andreancardona — APPROVED** [review]

**Pull request merged**